livekit-agents/livekit/agents/pipeline/room_io.py

```diff
@@ -184,29 +238,92 @@ async def _capture_text():
         self._tasks.add(task)
         task.add_done_callback(self._tasks.discard)
 
-    async def wait_for_participant(self) -> rtc.RemoteParticipant:
-        await self._participant_ready.wait()
-        assert self._participant is not None
-        return self._participant
-
     async def aclose(self) -> None:
         if self._closed:
             raise RuntimeError("RoomInput already closed")
 
         self._closed = True
         self._room.off("participant_connected", self._on_participant_connected)
-        self._participant = None
+        self._room.off("participant_disconnected", self._on_participant_disconnected)
+
+        self._active_participant = None
+        for name in list(self._streams.keys()):
+            if self._streams[name] is not None:
+                await self._streams[name].aclose()
+                self._streams[name] = None
 
         # cancel and wait for all pending tasks
         await utils.aio.cancel_and_wait(*self._tasks)
         self._tasks.clear()
 
-        if self._audio_stream is not None:
-            await self._audio_stream.aclose()
-            self._audio_stream = None
-        if self._video_stream is not None:
-            await self._video_stream.aclose()
-            self._video_stream = None
+    def _start_stream(self, *, audio: bool = True, video: bool = True) -> None:
+        self._close_stream(audio=audio, video=video)
```

Comment:
I think we need to clear out both audio and video here. it wouldn't make sense to handle audio from one user, but video from another